internal/logic/logic.go

```diff
@@ -49,5 +63,185 @@ func (l htmlPdfServiceLogic) Ping(req *model.PingRequest) *respModel.Response {
+			Status:  http.StatusInternalServerError,
+			Message: codes.GetErr(codes.ErrFileStoreFail),
+			Data:    nil,
+		}
+	}
+	return &respModel.Response{
+		Status:  http.StatusCreated,
+		Message: "SUCCESS",
+		Data: map[string]interface{}{
+			"id": u,
+		},
+	}
+}
+
+func (l htmlPdfServiceLogic) Replace(id string, file io.Reader) *respModel.Response {
+	_, err := l.dsSvc.GetFile(id)
+	if err != nil {
+		log.Error(err)
+		return &respModel.Response{
+			Status:  http.StatusInternalServerError,
+			Message: codes.GetErr(codes.ErrKeyNotFound),
+			Data:    nil,
+		}
+	}
+	fileBytes, err := ioutil.ReadAll(file)
+	if err != nil {
+		log.Error(err)
+		return &respModel.Response{
+			Status:  http.StatusInternalServerError,
+			Message: codes.GetErr(codes.ErrReadFileFail),
+			Data:    nil,
+		}
+	}
+	jb, err := l.htSvc.GetJsonFromHtml(fileBytes)
+	if err != nil {
+		return &respModel.Response{
+			Status:  http.StatusInternalServerError,
+			Message: codes.GetErr(codes.ErrFileConversionFail),
+			Data:    nil,
+		}
+	}
+	err = l.dsSvc.SaveFile(id, jb, 0)
+	if err != nil {
+		log.Error(err)
+		return &respModel.Response{
+			Status:  http.StatusInternalServerError,
+			Message: codes.GetErr(codes.ErrFileStoreFail),
+			Data:    nil,
+		}
+	}
+	return &respModel.Response{
+		Status:  http.StatusOK,
+		Message: "SUCCESS",
+		Data: map[string]interface{}{
+			"id": id,
+		},
+	}
+}
+
+func (l htmlPdfServiceLogic) HtmlToPdf(w io.Writer, req *model.GenerateReq) *respModel.Response {
+	var z map[string]interface{}
+	b, err := l.dsSvc.GetFile(req.Id)
+	if err != nil {
+		log.Error(err)
+		return &respModel.Response{
+			Status:  http.StatusInternalServerError,
+			Message: codes.GetErr(codes.ErrFetchingFile),
+			Data:    nil,
+		}
+	}
+	err = json.NewDecoder(bytes.NewBuffer(b)).Decode(&z)
+	if err != nil {
+		log.Error("error unmarshalling JSON:" + err.Error())
+		return &respModel.Response{
+			Status:  http.StatusInternalServerError,
+			Message: codes.GetErr(codes.ErrFileParseFail),
+			Data:    nil,
+		}
+	}
+	k, ok := z["Pages"].([]interface{})
+	if !ok {
+		log.Error("assertion for Pages failed")
+		return &respModel.Response{
+			Status:  http.StatusBadRequest,
+			Message: codes.GetErr(codes.ErrDecodingData),
+			Data:    nil,
+		}
+	}
+	for i, p := range k {
+		page, ok := p.(map[string]interface{})
+		if !ok {
+			log.Error("assertion for map[string]interface{} failed")
+			return &respModel.Response{
+				Status:  http.StatusBadRequest,
+				Message: codes.GetErr(codes.ErrDecodingData),
+				Data:    nil,
+			}
+		}
+		l, ok := page["Base64PageData"].(string)
+		if !ok {
+			continue
+		}
+		buf, err := base64.StdEncoding.DecodeString(l)
+		if err != nil {
+			log.Error("error decoding base 64 input on page " + fmt.Sprint(i) + " " + err.Error())
+			return &respModel.Response{
+				Status:  http.StatusInternalServerError,
+				Message: codes.GetErr(codes.ErrDecodingData),
+				Data:    nil,
+			}
+		}
+		log.Error(fmt.Sprintf("%+v", *req))
```

Comment:
Remove this error log...probably added for debugging.

Reply:
done

Reply:
Remove this..looks like this is commented.

Reply:
done